database/skylink.go

```diff
@@ -63,7 +69,11 @@ func (db *DB) SkylinkCreate(ctx context.Context, sl string, server string) (Skyl
 }
 
 // SkylinkFetch fetches a skylink from the DB.
-func (db *DB) SkylinkFetch(ctx context.Context, sl string) (Skylink, error) {
+func (db *DB) SkylinkFetch(ctx context.Context, skylink string) (Skylink, error) {
```

Comment:
It leaves quite a bit of ambiguity about the behaviour. Does it fetch one? Does it create it? You can't easily tell at a glance without knowing how the DB module works.

I don't mind renaming it, if you'd like it better. I'd prefer to do it in #7, if we're going it.

Reply:
I usually model the database interface very closely to the underlying driver or database technology. So in this case I would go for `FindSkylink`. 

I think I agree with Ivo when it comes to just calling it `Skylink`. I would not be sure whether it would create the Skylink if it does not exist, or if I would get a Skylink object that hasn't been persisted yet, or maybe something else even.

I don't like that `Fetch` comes after `Skylink`, dogs fetch 😛

Reply:
The reason it's `Skylink<action>` is so you can type `db.Skyl` and press your auto-complete shortcut and get a list of all available actions on skylinks.

Regarding the term - in DB systems `fetch` and `query` are used more or less interchangeably. MySQL and Postgres use `fetch`, DynamoDB uses `query`, and MongoDB apparently uses `find`. I guess what we find more logical depends on which DB we started with or used for longest time. I don't mind renaming this to `Find` but I typically expect `find` operations to return a list and not a single result. I'm not naming this `FindOneSkylink`, though.

Reply:
I don't think there would be a ton of confusion because that's just idiomatic getter/setter naming in Go. You want a skylink call it `Skylink`. You want to set one you call `SetSkylink`. You want to create one call `NewSkylink`.

If you'd like to put more emphasis on the fact that it involves I/O from a Mongo database then I'd go straight for `FindSkylink` to group all the operations that fetch data together and also to adhere to the mongo naming conventions. I generally find it more useful to group by action when using a library I don't know yet.
One example would be `binary.LittleEndian.PutUint64` / `binary.LittleEndian.Uint64`. Seems easier for me to type the operation I want and then see the variations than the other way round.

Reply:
We seem to be discussing two separate things here, so let's lay them out.

1. Order of "action" and "subject" when naming, e.g. "CreateSkylink" vs "SkylinkCreate".

My argument for "<subject><action>" is the ease of using auto-complete to list all available actions on a given subject. This is helpful for people who are unfamiliar with the service and want to achieve something.

That said, if we have stronger support for "<action><subject>" I won't block that and I'll refactor the service. I just find `AddServerSkylink`/`ServerAddSkylink`/`AddSkylinkServer` to be awkward and confusing - I will need to consult the docs before using that function because the name is not clear enough.

2. Action names - `set`/`new`/`create`.

If we want to go for standard go naming (new, get, set, update), I can agree to that. I'm not sure it always makes perfect sense when dealing with external data stores but it's acceptable.

I wouldn't go for Mongo-specific method naming because underlying technologies such as DB implementation should be replaceable without rewriting more than the persistence layer of the service. I don't consider either `find` or `fetch` to be technology specific in this sense, what I have in mind here is more stuff like `findOne` and `findAndUpdate`.

In any case, this doesn't seem like a blocking discussion and it also seems like a long conversation that should happen on Discord, rather than GitHub. To that end, I'll open a new thread on `#core-dev` and once we have a decision I'll refactor the service accordingly.